src/Microsoft.AspNet.Mvc.Core/ApplicationModels/AuthorizationApplicationModelProvider.cs

```diff
@@ -44,6 +44,11 @@ public void OnProvidersExecuting(ApplicationModelProviderContext context)
                     controllerModel.Filters.Add(new AuthorizeFilter(policy));
                 }
 
+                foreach (var attribute in controllerModel.Attributes.OfType<AllowAnonymousAttribute>())
```

Comment:
We can do that. In that case, should this https://github.com/aspnet/Mvc/pull/3330/files#diff-989bc8123071d2e7f36cc5d32dc275ffL41 also be changed to use its interface `IAuthorizeData`?